content/mondoo-linux-security.mql.yaml

```diff
@@ -5001,78 +4942,93 @@ queries:
+        )
+      auditd.rules.files.contains(path == "/etc/issue" && permissions == "wa" && keyname == "system-locale")
+      auditd.rules.files.contains(path == "/etc/issue.net" && permissions == "wa" && keyname == "system-locale")
+      auditd.rules.files.contains(path == "/etc/hosts" && permissions == "wa" && keyname == "system-locale")
   - uid: mondoo-linux-security-discretionary-access-control-permission-modification-events-are-collected
     title: Ensure discretionary access control permission modification events are audited
     impact: 50
-    props:
-      - uid: mondooLinuxSecurityAuditFiles
-        title: Return the content from all /etc/audit/rules.d and /etc/audit/audit.rules
-        mql: |
-          mondooLinuxSecurityAuditFiles = files.find(from: "/etc/audit/rules.d",regex:'.*\.rules$', type: "file").list.map(path) + ["/etc/audit/audit.rules"]
-          return mondooLinuxSecurityAuditFiles.map(file(_).content.lines.where( _ == /^[^#]/ ))
     mql: |
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /chmod/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /fchmod/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /fchmodat/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /chown/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /fchown/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /fchownat/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /lchown/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /setxattr/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /lsetxattr/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /fsetxattr/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /removexattr/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /lremovexattr/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.any(_ == /fremovexattr/)
-      props.mondooLinuxSecurityAuditFiles.flat.unique.where(_ == /chmod|fchmod|fchmodat|chown|fchown|fchownat|lchown|setxattr|lsetxattr|fsetxattr|removexattr|lremovexattr|fremovexattr/).all(
-        split("=").any(split(" ").any(_ == logindefs.params.UID_MIN))
-          && split("-").containsAll(["a always,exit ",])
-          && split("-").containsAll(["F auid!=","1 "])
-            || split("-").containsAll(["F auid!=unset "])
-            || split("-").containsAll(["F auid!=4294967295 "])
-          && split("-").containsAll(["F arch=b64 "])
-            || split("-").containsAll(["F arch=b32 "])
-          && split("-").containsAll(["F key=perm_mod"])
-            || split("-").containsAll(["k perm_mod"])
+      auditd.rules.syscalls.contains(
+        action == "always" &&
+        list == "exit" &&
+        fields.contains(key == "arch" && op == "=" && value == "b64") &&
+        fields.contains(key == "auid" && op == ">" && value == "=1000") &&
```

Comment:
The splitting needs to get fixed in the resource, but I'm leaving it here for now as this is the way the resource returns the data.

Reply:
Fixed in https://github.com/mondoohq/cnquery/pull/5858